Menus/ScenarioActivityConfigGUI.h

```diff
@@ -0,0 +1,149 @@
+#ifndef _RTESCENARIOACTIVITYCONFIGGUI_
+#define _RTESCENARIOACTIVITYCONFIGGUI_
+
+#include "Activity.h"
+
+namespace RTE {
+
+	class GameActivity;
+	class AllegroBitmap;
+	class GUIControlManager;
+	class GUICollectionBox;
+	class GUIComboBox;
+	class GUICheckbox;
+	class GUIButton;
+	class GUILabel;
+	class GUISlider;
+	class GUIEvent;
+
+	/// <summary>
+	/// Handling for the scenario Activity configuration screen composition and interaction.
+	/// </summary>
+	class ScenarioActivityConfigGUI {
+
+	public:
+
+#pragma region Creation
+		/// <summary>
+		/// Constructor method used to instantiate a ScenarioActivityConfigGUI object in system memory and make it ready for use.
+		/// </summary>
+		/// <param name="parentControlManager">Pointer to the parent GUIControlManager which owns all the GUIControls of this ScenarioActivityConfigGUI.</param>
+		explicit ScenarioActivityConfigGUI(GUIControlManager *parentControlManager);
+#pragma endregion
+
+#pragma region Getters and Setters
+		/// <summary>
+		/// Gets whether this Activity configuration menu is currently visible and enabled.
+		/// </summary>
+		/// <returns></returns>
+		bool IsEnabled() const;
+
+		/// <summary>
+		/// Enables or disables the Activity configuration menu.
+		/// </summary>
+		/// <param name="enable">Show and enable or hide and disable the Activity configuration menu.</param>
+		/// <param name="selectedActivity">Pointer to the Activity this ScenarioActivityConfigGUI will be configuring for.</param>
+		/// <param name="selectedScene">Pointer to the Scene the passed in Activity will be using.</param>
+		void SetEnabled(bool enable, const Activity *selectedActivity = nullptr, Scene *selectedScene = nullptr);
+#pragma endregion
+
+#pragma region Concrete Methods
+		/// <summary>
+		/// Updates the ScenarioActivityConfigGUI state.
+		/// </summary>
+		/// <param name="mouseX">Mouse X position.</param>
+		/// <param name="mouseY">Mouse Y position.</param>
+		bool Update(int mouseX, int mouseY);
+
+		/// <summary>
+		/// Draws the ScenarioActivityConfigGUI to the screen.
+		/// </summary>
+		void Draw();
+#pragma endregion
+
+	private:
+
+		/// <summary>
+		/// Enumeration for all the player columns in the player setup box. "Extends" the Players enumeration by adding an entry for the CPU player.
+		/// </summary>
+		enum PlayerColumns { PlayerCPU = Players::MaxPlayerCount, PlayerColumnCount };
```

Comment:
Don't use class enums when using to access index.

Reply:
You wanna avoid static casts cause you're always using it as an int? Or am I misunderstanding you?

Reply:
Yes.
Serves no benefit here.
And even though you can specify the underlaying type, you still need to explicitly cast.

Reply:
Hmmm, yeah, I'm a bit torn on this I guess. I do find it a pain to have to cast to int all the time, but I also am in favour of consistency and minimizing sonarlint complaints.
Leave it be I suppose, but we should try not to have too many of these non-class enums.

Reply:
I don't agree with `we should try not to have too many of these non-class enums.`
Regular enums aren't suddenly a crime against humanity. We should use the right thing for the task.
Besides, private enums are inaccessible to non-friend classes so they don't pollute whatever the class is included in anyway.

Reply:
But muh sonarlint!
Yeah fair enough, if they're private they're not gonna be spilling out anywhere, so that's a good point. It is still better to use them than to not, explicit casting from class enums means you can't compare 2 unrelated enums (which would be bad), but you're right that sometimes unscoped C-style enums are the more appropriate tool.

Reply:
When Bjarne:tm: implements the use of class enums with specified int underlaying type without explicit casting for array index access in C++9001, then we can get rid of regular enums.

Reply:
lol I appreciate the optimism but I think you're missing a 0 in there.
Anyway, feel free to resolve.